### DISCUSSION TITLE
Load/Mixed is not registered and is not a subclass of Initializer

### DISCUSSION THREAD

**Discussion by squidszyd** [apache/incubator-mxnet#14792] [Q&A]
I'm trying to use [`mx.init.Load`](https://mxnet.incubator.apache.org/versions/master/api/python/optimization/optimization.html#mxnet.initializer.Load) initializer to define the source of parameter when declaring some `Variable`, say `my_var`. However,  during the construction of `my_var`, an `AttributeError` is thrown [here](https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/symbol/symbol.py#L2637), saying `'Load' object has no attribute 'dumps'`. By checking the declaration of `mx.init.Load`, I found that only `mx.init.Load` and `mx.init.Mixed` are declared without registering and subclassing:
```
# declaration of Load:
class Load(object):
  def __init__(...):
    # NOT a subclass of initializer and NO super(...) here!
    ...

# declaration of Mixed:
class Mixed(object):
  def __init__(...):
    # NOT a subclass of initializer and NO super(...) here!
    ...

# declaration of Ones and Zeros
@register
@alias("ones")
class One(Initializer):
  def __init__(self):
    super(One, self).__init__()
    ...

@register
@alias("zeros")
class Zero(Initializer):
  def __init__(self):
    super(Zero, self).__init__()
    ...
```

Thus, neither `mx.init.Load` nor `mx.init.Mixed` will share the functionality `dumps()` like those  subclasses of `Initializer`.

I think this is the core cause of the above mentioned exception. But I'm not sure if this is designed for purpose?

*EDIT-Apr.25*
There is another issue with `mx.init.Constant(...)`. The documentation has noted that user can pass an `NDArray` to its constructor. However, `NDArray` is not JSON-serializable:
```
>>> x = mx.nd.random.normal(shape=(3, 5))
>>> x
[[ 2.21220636  0.7740038   1.04344046  1.18392551  1.89171135]
 [-1.23474145 -1.771029   -0.45138445  0.57938355 -1.85608196]
 [-1.9768796  -0.20801921  0.2444218  -0.03716067 -0.48774993]]
<NDArray 3x5 @cpu(0)>
>>> y = mx.sym.Variable("y", init=mx.init.Constant(x))
>>> 
...
TypeError:
[[ 2.21220636  0.7740038   1.04344046  1.18392551  1.89171135]
 [-1.23474145 -1.771029   -0.45138445  0.57938355 -1.85608196]
 [-1.9768796  -0.20801921  0.2444218  -0.03716067 -0.48774993]]
<NDArray 3x5 @cpu(0)> is not JSON serializable
```

I've figured out a workaround that converts `NDArray` to `List`, i.e. `x.asnumpy().tolist()`. Please update the document for this usage.

**Comment by piiswrong** [comment]
Load can only be used as a global initializer for loading previously saved models.
